pkg/expr/service.go

```diff
@@ -34,9 +36,13 @@ const DatasourceID = -100
 // should be used instead and should be set to "__expr__".
 const OldDatasourceUID = "-100"
 
-// IsDataSource checks if the uid points to an expression query
-func IsDataSource(uid string) bool {
-	return uid == DatasourceUID || uid == OldDatasourceUID
+// QueryKindByDatasourceUID returns NodeType depending on the UID of the data source: TypeCMDNode if UID is DatasourceUID
+// or OldDatasourceUID, and TypeDatasourceNode otherwise.
+func QueryKindByDatasourceUID(uid string) NodeType {
```

Comment:
Rename to `NodeTypeFromDatasourceUID` (or `NodeTypeByDatasourceUID`). `NodeType` matches the return type so that is what I mostly care about. "From" just sounds better to be then "By", but don't care about that is much.